lib/features/auth/login/login_page.dart

```diff
@@ -20,36 +20,53 @@ class LoginPage extends StatefulWidget with HasPresenter<LoginPresenter> {
   State<LoginPage> createState() => _LoginPageState();
 }
 
-class _LoginPageState extends State<LoginPage> with PresenterStateMixin<LoginViewModel, LoginPresenter, LoginPage> {
+class _LoginPageState extends State<LoginPage>
+    with PresenterStateMixin<LoginViewModel, LoginPresenter, LoginPage> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) => Scaffold(
-        body: Padding(
-          padding: const EdgeInsets.all(32.0),
-          child: Column(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: [
-              TextField(
-                decoration: InputDecoration(
-                  hintText: appLocalizations.usernameHint,
+        body: stateObserver(
+          builder: (context, state) => Padding(
+            padding: const EdgeInsets.all(
+              Dimensions.PADDING_32,
+            ),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                TextField(
+                  decoration: InputDecoration(
+                    hintText: appLocalizations.usernameHint,
+                  ),
+                  onChanged: (username) => presenter.usernameChanged(username),
+                ),
+                const SizedBox(
+                  height: Dimensions.ITEM_HEIGHT_8,
+                ),
+                TextField(
+                  obscureText: true,
+                  decoration: InputDecoration(
+                    hintText: appLocalizations.passwordHint,
+                  ),
+                  onChanged: (password) => presenter.passwordChanged(password),
                 ),
-                onChanged: (text) => doNothing(), //TODO
-              ),
-              const SizedBox(height: 8),
-              TextField(
-                obscureText: true,
-                decoration: InputDecoration(
-                  hintText: appLocalizations.passwordHint,
+                const SizedBox(
+                  height: Dimensions.ITEM_HEIGHT_16,
                 ),
-                onChanged: (text) => doNothing(), //TODO
-              ),
-              const SizedBox(height: 16),
-              stateObserver(
-                builder: (context, state) => ElevatedButton(
-                  onPressed: () => doNothing(), //TODO
-                  child: Text(appLocalizations.logInAction),
+                stateObserver(
+                  builder: (context, state) => state.ifValid
+                      ? ElevatedButton(
+                          onPressed: () => presenter.login(),
+                          child: Text(appLocalizations.logInAction),
+                        )
+                      : const SizedBox.shrink(),
```

Comment:
hiding button is a bit agains the good UX practices, we should either disable it or show snackbar/toast on click stating the username/password is invalid

Reply:
Yeah, I understand it! I thought about disabling as about removing, but it's better to use your approach
I will fix it soon